packages/go_router/lib/src/delegate.dart

```diff
@@ -116,10 +119,6 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       final RouteBase route = match.route;
       if (route is GoRoute && route.parentNavigatorKey != null) {
         final bool canPop = route.parentNavigatorKey!.currentState!.canPop();
-        // Similar to popRoute, it should not be possible for a GoRoute with
-        // parentNavigatorKey to be the only page, so canPop should return true
-        // in this case.
-        assert(canPop);
         return canPop;
```

Comment:
This should continue if the canpop is false.

Reply:
Good catch, fixed: 15171e26bea3e4e750e3870505c523c8ba4a2e6c